rust-src/crypto_common/src/encryption.rs

```diff
@@ -23,25 +23,27 @@ impl From<String> for Password {
 impl FromStr for Password {
     type Err = <String as FromStr>::Err;
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> { Ok(Password { password: s.into() }) }
+    fn from_str(s: &str) -> ParseResult<Self, Self::Err> { Ok(Password { password: s.into() }) }
```

Comment:
The type of `from_str` is just `Result`, so using `ParseResult` here is not a good idea. This comes from the standard library.